File: exercises/exercise-1-testing-java/src/main/java/de/emaarco/example/application/port/inbound/bike/StartBikeSubscriptionUseCase.java
package de.emaarco.example.application.port.inbound.bike;

import de.emaarco.example.domain.Email;
import de.emaarco.example.domain.Name;
import de.emaarco.example.domain.bike.BikeId;
import de.emaarco.example.domain.bike.BikeSubscriptionId;

public interface StartBikeSubscriptionUseCase {
    record Command(
        BikeId bikeId,
        Email email,
        Name name
    ) {}

    BikeSubscriptionId start(Command command);
}

File: exercises/exercise-1-testing-kotlin/src/main/kotlin/de/emaarco/example/application/port/inbound/bike/StartBikeSubscriptionUseCase.kt
package de.emaarco.example.application.port.inbound.bike

import de.emaarco.example.domain.Email
import de.emaarco.example.domain.Name
import de.emaarco.example.domain.bike.BikeId
import de.emaarco.example.domain.bike.BikeSubscriptionId

interface StartBikeSubscriptionUseCase {
    data class Command(
        val bikeId: BikeId,
        val email: Email,
        val name: Name
    )

    fun start(command: Command): BikeSubscriptionId
}
